app/connectors/CreateIncomeSourceConnector.scala

```diff
@@ -31,45 +31,66 @@ class CreateIncomeSourceConnector @Inject()(val http: HttpClient,
-            Left(CreateBusinessErrorResponse(response.status, s"Not valid json: ${response.body}"))
+            Logger("application").error(s"[CreateIncomeSourceConnector][createBusiness] - Json validation error parsing business income sources response, error ${response.body}")
+            Left(CreateIncomeSourcesErrorResponse(response.status, s"Not valid json: ${response.body}"))
           },
           valid =>
             Right(valid)
         )
       case response =>
         Logger("application").error(s"[CreateIncomeSourceConnector][createBusiness] - Response status: ${response.status}, body: ${response.body}")
-        Left(CreateBusinessErrorResponse(response.status, s"Error creating incomeSource: ${response.json}"))
+        Left(CreateIncomeSourcesErrorResponse(response.status, s"Error creating incomeSource: ${response.json}"))
     }
   }
 
   def createForeignProperty(mtdItid: String, createForeignPropertyRequest: CreateForeignPropertyIncomeSource)
-                           (implicit headerCarrier: HeaderCarrier): Future[Either[CreateBusinessErrorResponse, List[AddIncomeSourceResponse]]] = {
+                           (implicit headerCarrier: HeaderCarrier): Future[Either[CreateIncomeSourcesErrorResponse, List[CreateIncomeSourcesResponse]]] = {
     val bodyAsJson = Json.toJson(createForeignPropertyRequest)
-    val url = addBusinessDetailsUrl(mtdItid)
+    val url = createBusinessIncomeSourcesUrl(mtdItid)
     http.POST(url, bodyAsJson).map {
       case response if response.status == OK =>
-        response.json.validate[List[AddIncomeSourceResponse]].fold(
+        response.json.validate[List[CreateIncomeSourcesResponse]].fold(
           _ => {
-            Logger("application").error(s"[CreateIncomeSourceConnector][createForeignProperty] - Json validation error parsing repayment response, error ${response.body}")
-            Left(CreateBusinessErrorResponse(response.status, s"Not valid json: ${response.body}"))
+            Logger("application").error(s"[CreateIncomeSourceConnector][createForeignProperty] - Json validation error parsing business income sources response, error ${response.body}")
+            Left(CreateIncomeSourcesErrorResponse(response.status, s"Not valid json: ${response.body}"))
           },
           valid =>
             Right(valid)
         )
       case response =>
         Logger("application").error(s"[CreateIncomeSourceConnector][createForeignProperty] - Response status: ${response.status}, body: ${response.body}")
-        Left(CreateBusinessErrorResponse(response.status, s"Error creating incomeSource: ${response.json}"))
+        Left(CreateIncomeSourcesErrorResponse(response.status, s"Error creating incomeSource: ${response.json}"))
+    }
+  }
+
+  def createUKProperty(mtdItid: String, createUKPropertyRequest: CreateUKPropertyIncomeSource)
```

Comment:
we now have 3 methods basically the same in this file (apart from logging text?) - can we unify them and make this file DRY?

Reply:
yeah not ideal, lunar implemented this - will fix